test/test-proxied-properties.js

```diff
@@ -0,0 +1,104 @@
+        const setResult2 = window.browser.nowrapns.nowrapkey = "new-value";
+
+        // Check the results of setting the new value of the wrapped properties.
+        equal(setResult, "new-value",
+                     "Got the expected result from setting a wrapped property name");
+        equal(setResult2, "new-value",
+                     "Got the expected result from setting a wrapped property name");
+
+        // Verify that the wrapped properties has been updated.
+        equal(window.browser.runtime.myprop, "new-value",
+                     "Got the expected updated value from the browser property");
+        equal(window.browser.nowrapns.nowrapkey, "new-value",
+                     "Got the expected updated value from the browser property");
+
+        // Verify that the target properties has been updated.
+        equal(window.chrome.runtime.myprop, "new-value",
+                     "Got the expected updated value on the related chrome property");
+        equal(window.chrome.nowrapns.nowrapkey, "new-value",
+                     "Got the expected updated value on the related chrome property");
+
+        // Set a property multiple times before read.
+        window.browser.nowrapns.nowrapkey2 = "new-value2";
+        window.browser.nowrapns.nowrapkey2 = "new-value3";
+
+        equal(window.chrome.nowrapns.nowrapkey2, "new-value3",
+                     "Got the expected updated value on the related chrome property");
+        equal(window.browser.nowrapns.nowrapkey2, "new-value3",
+                     "Got the expected updated value on the wrapped property");
+      });
+    });
+
+    it("deletes proxy getter/setter that are not wrapped", () => {
+      const fakeChrome = {};
+      return setupTestDOMWindow(fakeChrome).then(window => {
+        window.browser.newns = {newkey: "test-value"};
+
+        ok("newns" in window.browser, "The custom namespace is in the wrapper");
+        ok("newns" in window.chrome, "The custom namespace is in the target");
+
+        equal(window.browser.newns.newkey, "test-value",
+                     "Got the expected result from setting a wrapped property name");
+
+        const setRes = window.browser.newns = {newkey2: "new-value"};
+        equal(window.browser.newns.newkey2, "new-value",
+                     "The new non-wrapped getter is cached");
+        deepEqual(setRes, {newkey2: "new-value"},
+                     "Got the expected result from setting a new wrapped property name");
+        deepEqual(window.browser.newns, window.chrome.newns,
+                         "chrome.newns and browser.newns are the same");
+
+        delete window.browser.newns.newkey2;
+        equal(window.browser.newns.newkey2, undefined,
+                     "Got the expected result from setting a wrapped property name");
+        ok(!("newkey2" in window.browser.newns),
+                  "The deleted property is not listed anymore");
+      });
```

Comment:
Eh, you get the picture. Align all of the ones between the last comment and this one. And in the other files.